src/state/surveys/surveys.reducer.ts

```diff
@@ -34,6 +35,38 @@ export function surveysReducer(state: SurveysState = initialSurveysState, action
                         : survey,
                 ),
             };
+
+        case SurveysActionType.RemoveSurveyItemAction:
+            return {
+                ...state,
+                surveys: state.surveys.map(survey =>
+                    survey.id === action.payload.surveyId
+                        ? {
+                              ...survey,
+                              items: [...survey.items.filter(item => item.id !== action.payload.surveyItem.id)],
+                          }
+                        : survey,
+                ),
+            };
+
+        case SurveysActionType.UpdateSurveyItemAction:
+            return {
+                ...state,
+                surveys: state.surveys.map(survey =>
```

Comment:
I wanna make some kind of adapter for this thing, this whole `state.surveys.map` dance is gonna get old soon I think 😛

Reply:
I’ll create an issue